cdap-ui/app/cdap/components/AppDetailedView/Tabs/HistoryTab.js

```diff
@@ -84,55 +105,56 @@ export default class HistoryTab extends Component {
           );
         });
   }
+
   componentWillUnmount() {
     this.pollSubscriptions
-        .forEach(subscription => {
-          subscription.dispose();
-        });
+      .forEach(subscription => {
+        subscription.dispose();
+      });
+  }
+
+  renderTableBody(history) {
+    let historyState = history || this.state.history;
+    return (
+      <tbody>
+        {
+          historyState
+            .map( history => {
+              return (
+                <tr key={history.runid}>
+                  <td>{history.programName}</td>
+                  <td>{humanReadableDate(history.start)}</td>
+                  <td>{history.runid}</td>
+                  <td>{history.status}</td>
+                  <td>
+                    <LogAction
+                      entity={{
+                        id: history.programName,
+                        uniqueId: history.runid,
+                        applicationId: history.appId,
+                        programType: history.programType
```

Comment:
this seems wrong.. I think we might need to update the LogAction fast action to accept RunId parameter. If runId is specified, just use that and do not poll for the run status.